modules/core/arrow-annotations-processor/src/main/java/arrow/generic/CoproductFileGenerator.kt

```diff
@@ -0,0 +1,194 @@
+                    .addModifiers(KModifier.SEALED)
+                    .addTypeVariables(generics.map { TypeVariableName(it) })
+                    .build()
+    )
+
+    for (generic in generics) {
+        addType(
+                TypeSpec.classBuilder(genericsToClassNames[generic]!!)
+                        .addModifiers(KModifier.INTERNAL, KModifier.DATA)
+                        .addTypeVariables(generics.toTypeParameters())
+                        .superclass(parameterizedCoproductNClassName(generics))
+                        .addProperty(
+                                PropertySpec.builder(generic.toLowerCase(), TypeVariableName(generic))
+                                        .initializer(generic.toLowerCase())
+                                        .build()
+                        )
+                        .primaryConstructor(
+                                FunSpec.constructorBuilder()
+                                        .addParameter(generic.toLowerCase(), TypeVariableName(generic))
+                                        .build()
+                        )
+                        .build()
+        )
+    }
+}
+
+private fun FileSpec.Builder.addCoproductOfConstructors(generics: List<String>) {
+    for (generic in generics) {
+        val additionalParameterCount = generics.indexOf(generic)
+
+        addFunction(
+                FunSpec.builder("coproductOf")
+                        .addAnnotations(additionalParameterSuppressAnnotation(additionalParameterCount))
+                        .addTypeVariables(generics.toTypeParameters())
+                        .addParameter(generic.toLowerCase(), TypeVariableName(generic))
+                        .addParameters(additionalParameterSpecs(additionalParameterCount))
+                        .addStatement("return ${genericsToClassNames[generic]}(${generic.toLowerCase()})")
+                        .returns(parameterizedCoproductNClassName(generics))
+                        .build()
+        )
+    }
+}
+
+private fun FileSpec.Builder.addCopExtensionConstructors(generics: List<String>) {
+    for (generic in generics) {
+        val additionalParameterCount = generics.indexOf(generic)
+
+        addFunction(
+                FunSpec.builder("cop")
+                        .addAnnotations(additionalParameterSuppressAnnotation(additionalParameterCount))
+                        .receiver(TypeVariableName(generic))
+                        .addTypeVariables(generics.toTypeParameters())
+                        .addParameters(additionalParameterSpecs(additionalParameterCount))
+                        .addStatement("return coproductOf<${generics.joinToString(separator = ", ")}>(this)")
+                        .returns(parameterizedCoproductNClassName(generics))
+                        .build()
+        )
+    }
+}
+
+private fun FileSpec.Builder.addSelectFunctions(generics: List<String>) {
+    addImport("arrow.core", "Option")
```

Comment:
For commonly used package like `arrow.core` would be nice to have constants or properties in some utilities so they are shared across the processors.
```
addImport(Packages.arrow.core)
```